File: chap20/ZeroDUneFonction.c
// Ces programmes sont sous licence CeCILL-B V1.
#include <stdio.h>
#include <math.h>

int main()
{
   double a, b, m, e;
   e = 1E-5;
   a = 2.0;
   b = 4.0;
   m = (a + b) / 2;
   while (b - a > e && fabs(sin(m)) > e) 
   {
      if (sin(a) * sin(m) <= 0) 
      {
         b = m;
      }   
      else
      { 
         a = m;
      }
      m = (a + b) / 2;
   }
   printf("%lf\n",m);
}

File: chap20/ZeroDUneFonction.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let _ =
  let seuil = 10. **(-5.) in
  let inf = ref 2. in
  let sup = ref 4. in
  let milieu = ref ((!inf +. !sup) /. 2.) in
  while !sup -. !inf > seuil && abs_float (sin !milieu) > seuil do
    if (sin !inf) *. (sin !milieu) <= 0. then
      sup := !milieu
    else
      inf := !milieu;
    milieu := (!inf +. !sup) /. 2.;
  done;
  print_float !milieu;
  print_newline ()

